mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -1047,6 +1076,12 @@ private[ml] object RandomForest extends Logging {
       numNodesInGroup += 1
       memUsage += nodeMemUsage
     }
+    if (memUsage > maxMemoryUsage && maxMemoryUsage != 0) {
```

Comment:
Thanks, I think I left that in from an earlier solution.

I agree it's a little odd to support 0, but honestly, I did it because it made writing one of the unit tests much easier.  The test for splitting a level of a tree in 1 group vs. multiple groups is pretty hard to write without it.